src/renderer/features/fellowship-votes/components/VotingHistoryList.tsx

```diff
@@ -0,0 +1,69 @@
+import { useUnit } from 'effector-react';
+import { memo, useMemo, useState } from 'react';
+
+import { type Chain } from '@/shared/core';
+import { useI18n } from '@/shared/i18n';
+import { useDeferredList } from '@/shared/lib/hooks';
+import { performSearch, toAddress } from '@/shared/lib/utils';
+import { FootnoteText, SearchInput } from '@/shared/ui';
+import { type Vote as VoteType } from '@/domains/collectives';
+import { identityModel } from '../model/identity';
+
+import { Vote } from './Vote';
+import { VotingHistoryListEmptyState } from './VotingHistoryListEmptyState';
+
+type Props = {
+  items: VoteType[];
+  chain: Chain | null;
+  loading?: boolean;
+};
+
+export const VotingHistoryList = memo(({ items, chain, loading }: Props) => {
+  const { t } = useI18n();
+  const [query, setQuery] = useState<string>('');
+
+  const identity = useUnit(identityModel.$identity);
+
+  const extendedItems = useMemo(() => {
+    return items.map(item => ({
+      ...item,
+      address: toAddress(item.accountId, { prefix: chain?.addressPrefix }),
+      name: identity[item.accountId].name ?? null,
+    }));
+  }, [items, identity]);
+
+  const filteredItems = useMemo(
+    () => performSearch({ records: extendedItems, query, weights: { address: 0.5, name: 1 } }),
+    [items, query],
+  );
+  const { list: deferredItems, isLoading } = useDeferredList({ list: filteredItems, isLoading: !!loading });
+
+  if (!chain) {
+    return null;
+  }
+
+  const shouldRenderLoader = isLoading;
```

Comment:
you can make an alias for isLoading the same way as for the list 
`const { list: deferredItems, isLoading: shouldRenderLoader} = useDeferredList({ list: filteredItems, isLoading: !!loading });`